arangod/Cluster/ClusterInfo.cpp

```diff
@@ -896,66 +898,80 @@ void ClusterInfo::loadPlan() {
 
         auto const collectionId = collectionPairSlice.key.copyString();
 
+        uint64_t computedHash = collectionSlice.hash();
```

Comment:
The `Slice::hash()` method is sensitive to different encoding types, e.g. object with index table/object without index table etc. This could be an issue if the slice is built by different places in the code that use different VelocyPack settings.
But as we are only checking for absolute equality here, this should be ok. In the worst case, we will rebuild the collection objects more often than necessary, but this isn't unsafe.

Reply:
Yes, this was my thinking, too. And the encoding is always the same, since the object comes the same way from the agency.